cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -351,6 +350,18 @@ public void onLost(Network network) {
       }
       broadcastVpnConnectivityChange(OutlinePlugin.ConnectionStatus.RECONNECTING);
     }
+
+    // Returns whether the underlying networks of NetworkInfo objects are equal.
+    private boolean networkEquals(final NetworkInfo a, final NetworkInfo b) {
```

Comment:
Yuck! There's no useful `NetworkInfo.equals()`? Is `Objects.equals` available to us?

Reply:
Unfortunately not. In my testing, the `NetworkInfo.equals` method always returns false, as if performing a reference comparison 😞.

Reply:
Damn! And `Object.equals` may not be available...?

Reply:
`Objects.equlals` does not solve the problem, as it determines equality by using the equals method of the first argument (when neither arguments are null).